java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightServerMiddleware.java

```diff
@@ -72,6 +74,10 @@
     }
   }
 
+  default Context onAuthenticationSuccess(Context currentContext) {
```

Comment:
We shouldn't re-export gRPC types in this module. Otherwise, this would just have been `io.grpc.ServerInterceptor`.

Reply:
Added RequestContext facade for this.